crypto/signatures.go

```diff
@@ -69,8 +72,47 @@ func VerifyHash(data Hash, pk PublicKey, sig Signature) error {
 	return nil
 }
 
+// WriteSignedObject writes a length-prefixed object followed by its signature.
+func WriteSignedObject(w io.Writer, obj interface{}, sk SecretKey) error {
+	encObj := encoding.Marshal(obj)
+	sig, _ := SignHash(HashBytes(encObj), sk) // no error possible
+	return encoding.NewEncoder(w).EncodeAll(encObj, sig)
```

Comment:
Is this the best way to do it? Also, I'm thinking it might cause problems. Intuitively, I would have expected WriteSignedObject to look like the object bytes followed by the signature bytes. But instead, we have a bytes slice which decodes to the object bytes followed by the signature bytes. You get an extra 8-byte prefix that I don't think needs to be there.